libs/agno/agno/agent/agent.py

```diff
@@ -398,16 +383,27 @@ def __init__(
         self.monitoring = monitoring
         self.telemetry = telemetry
 
-        self.run_id = None
-        self.run_input = None
-        self.run_messages = None
-        self.run_response = None
-        self.images = None
-        self.videos = None
-        self.audio = None
+        # --- Params not to be set by user ---
+        self.session_metrics: Optional[SessionMetrics] = None
 
-        self.agent_session = None
-        self._formatter = None
+        self.run_id: Optional[str] = None
+        self.run_input: Optional[Union[str, List, Dict, Message]] = None
+        self.run_messages: Optional[RunMessages] = None
+        self.run_response: Optional[RunResponse] = None
+
+        # Images generated during this session
+        self.images: Optional[List[ImageArtifact]] = None
+        # Videos generated during this session
+        self.videos: Optional[List[VideoArtifact]] = None
+        # Audio generated during this session
+        self.audio: Optional[List[AudioArtifact]] = None
+        # Agent session
+        self.agent_session: Optional[AgentSession] = None
+
+        self.tools_for_model: Optional[List[Dict]] = None
```

Comment:
Does this need to be a root-level attribute? I think it can just be local in `add_tools_to_model`